channeldb/payments.go

```diff
@@ -530,6 +570,45 @@ func deserializeHop(r io.Reader) (*route.Hop, error) {
 		return nil, err
 	}
 
+	// TODO(roasbeef): change field to allow LegacyPayload false to be the
+	// legacy default?
+	err := binary.Read(r, byteOrder, &h.LegacyPayload)
+	if err != nil {
+		return nil, err
+	}
+
+	var numElements uint32
+	if err := ReadElements(r, &numElements); err != nil {
+		return nil, err
+	}
+
+	// If there're no elements, then we can return early.
+	if numElements == 0 {
+		return h, nil
+	}
+
+	tlvMap := make(map[uint64][]byte)
+	for i := uint32(0); i < numElements; i++ {
+		var tlvType uint64
+		if err := ReadElements(r, &tlvType); err != nil {
+			return nil, err
+		}
+
+		rawRecordBytes, err := wire.ReadVarBytes(r, 0, 1300, "tlv")
```

Comment:
`sphinx.MaxPayloadSize` ?

Reply:
That's slightly different, this is mean to just be a ball park estimate. We also can't import that package here, as otherwise, it'll cause an import cycle.

Reply:
Then at least make it a local constant with a name. 

I was able to import sphinx in this file, no circular dependency by the way.

Reply:
Ah actually [we moved the `channeldb` items out of the onion package a while ago to address this](https://github.com/lightningnetwork/lightning-onion/pull/22). Opted to define a new local variable so we don't need to add the `sphinx` package to the dependency graph for this package just to have access to a constant.